### PR TITLE
Add docstrings to conftest files

### LINKED ISSUE #300: Missing docstrings
## Description
- write any missing docstrings

## User Story
As an engineer it is important to hand off work which has proper documentation and descriptions.

### PR DESCRIPTION
**Pull request by annalisev**
I've added docstrings to the conftest files.

Let me know if any can be updated.

Closes #300

### CODE REVIEW

File: pipeline/extract/epic/conftest.py
Comment (hhm970) [inline pipeline/extract/epic/conftest.py:1]: A thought I've had (from doing the `conftest.py` corresponding to `test_load.py`); do we need type hints for the tests? (ie. saying what type of object is outputted from the fixtures)

File: pipeline/extract/epic/conftest.py [hunk 1]
```diff
@@ -1,8 +1,11 @@
+'''A conftest file that defines functions used in test_epic.py.'''
+
 import pytest
 
 
 @pytest.fixture
 def epic_game_post_response():
```
Comment (shindym) [inline pipeline/extract/epic/conftest.py:7]: Add typehints?

File: pipeline/extract/epic/conftest.py
Comment (annalisev) [reply pipeline/extract/epic/conftest.py:1]: Yes good point I will add typehints

File: pipeline/extract/epic/conftest.py (hunk 1 above)
Comment (annalisev) [inline pipeline/extract/epic/conftest.py:7]:
```suggestion
def epic_game_post_response() -> Dict:
```

File: pipeline/extract/epic/conftest.py [hunk 2]
```diff
@@ -64,6 +67,7 @@ def epic_game_post_response():
 
 @pytest.fixture
 def epic_game_data_game_obj():
```
Comment (annalisev) [inline pipeline/extract/epic/conftest.py:69]:
```suggestion
def epic_game_data_game_obj() -> Dict:
```

File: pipeline/extract/epic/conftest.py [hunk 3]
```diff
@@ -116,6 +120,7 @@ def epic_game_data_game_obj():
 
 @pytest.fixture
 def epic_empty_post_request():
```
Comment (annalisev) [inline pipeline/extract/epic/conftest.py:122]:
```suggestion
def epic_empty_post_request() -> Dict:
```

File: pipeline/extract/gog/conftest.py [hunk 4]
```diff
@@ -1,10 +1,13 @@
+'''A conftest file that defines functions used in test_extract_gog.py.'''
+
 import pytest
 
 from bs4 import BeautifulSoup
 
 
 @pytest.fixture
 def gog_fake_links():
```
Comment (annalisev) [inline pipeline/extract/gog/conftest.py:9]:
```suggestion
def gog_fake_links() -> list[BeautifulSoup]:
```

File: pipeline/extract/gog/conftest.py [hunk 5]
```diff
@@ -14,6 +17,7 @@ def gog_fake_links():
 
 @pytest.fixture
 def gog_fake_links_bad():
```
Comment (annalisev) [inline pipeline/extract/gog/conftest.py:19]:
```suggestion
def gog_fake_links_bad() -> list[BeautifulSoup]:
```

File: pipeline/extract/gog/conftest.py [hunk 6]
```diff
@@ -13,42 +16,48 @@ def gog_fake_links():
...
-def gog_tags():
+def gog_tags() -> list[BeautifulSoup]:
+    '''Returns an example of a Soup of game tags.'''
     return BeautifulSoup(
         '''<span class="details__link-text">tag7</span>
         <span class="details__link-text">tag72</span>''', features='html.parser')
 
 
 @pytest.fixture
 def gog_tags_bad():
```
Comment (annalisev) [inline pipeline/extract/gog/conftest.py:36]:
```suggestion
def gog_tags_bad() -> list[BeautifulSoup]:
```

File: pipeline/extract/gog/conftest.py [hunk 7]
```diff
@@ -13,42 +16,48 @@ def gog_fake_links():
...
-def gog_tags_bad():
+def gog_tags_bad() -> list[BeautifulSoup]:
+    '''Returns a bad example of a Soup of game tags.'''
     return BeautifulSoup(
         '''<span class="detail">tag7</span>
         <span class="detailt">tag72</span>''', features='html.parser')
 
 
 @pytest.fixture
 def gog_title():
```
Comment (annalisev) [inline pipeline/extract/gog/conftest.py:44]:
```suggestion
def gog_title() -> list[BeautifulSoup]:
```

File: pipeline/extract/gog/conftest.py [hunk 8]
```diff
@@ -13,42 +16,48 @@ def gog_fake_links():
...
+def gog_tags() -> list[BeautifulSoup]:
+    '''Returns an example of a Soup of game tags.'''
     return BeautifulSoup(
         '''<span class="details__link-text">tag7</span>
         <span class="details__link-text">tag72</span>''', features='html.parser')
 
 
 @pytest.fixture
-def gog_tags_bad():
+def gog_tags_bad() -> list[BeautifulSoup]:
```
Comment (annalisev) [inline pipeline/extract/gog/conftest.py:36]:
```suggestion
def gog_tags_bad() -> BeautifulSoup:
```

File: pipeline/extract/gog/conftest.py [hunk 9]
```diff
@@ -13,42 +16,48 @@ def gog_fake_links():
...
+    '''Returns badly formatted hrefs.'''
     return [BeautifulSoup(
         '<a href="/gamesubliesda">publisher1</a>', features='html.parser').find('a'), BeautifulSoup(
         '<a href="/gam">seven</a>', features='html.parser').find('a'), BeautifulSoup(
         '<a href="/shers=Bethesda">word2</a>', features='html.parser').find('a')]
 
 
 @pytest.fixture
-def gog_tags():
+def gog_tags() -> list[BeautifulSoup]:
```
Comment (annalisev) [inline pipeline/extract/gog/conftest.py:28]:
```suggestion
def gog_tags() -> BeautifulSoup:
```

File: pipeline/extract/gog/conftest.py [hunk 10]
```diff
@@ -13,42 +16,48 @@ def gog_fake_links():
...
-def gog_title():
+def gog_title() -> BeautifulSoup:
+    '''Returns an example of a game title.'''
     return BeautifulSoup(
         '''<div class="product-tile__title" title='goodtitle'>tag7</div>
         <span class="details__link-text">tag72</span>''', features='html.parser')
 
 
 @pytest.fixture
 def gog_description():
```
Comment (annalisev) [inline pipeline/extract/gog/conftest.py:52]:
```suggestion
def gog_description() -> BeautifulSoup:
```

File: pipeline/extract/gog/conftest.py [hunk 11]
```diff
@@ -13,42 +16,48 @@ def gog_fake_links():
...
 @pytest.fixture
-def gog_description():
+def gog_description() -> BeautifulSoup:
+    '''Returns an example of a game description.'''
     return BeautifulSoup(
         '''<div class="description" >a very nice really thorough description </div>''', features='html.parser')
 
 
 @pytest.fixture
 def gog_time():
```
Comment (annalisev) [inline pipeline/extract/gog/conftest.py:59]:
```suggestion
def gog_time() -> BeautifulSoup:
```

File: pipeline/extract/steam/conftest.py [hunk 12]
```diff
@@ -6,6 +6,7 @@
 
 @pytest.fixture
 def steam_rating_soup():
```
Comment (annalisev) [inline pipeline/extract/steam/conftest.py:8]:
```suggestion
def steam_rating_soup() -> BeautifulSoup:
```

File: pipeline/extract/steam/conftest.py [hunk 13]
```diff
@@ -14,34 +15,41 @@ def steam_rating_soup():
 
 @pytest.fixture
 def steam_no_rating_soup():
```
Comment (annalisev) [inline pipeline/extract/steam/conftest.py:17]:
```suggestion
def steam_no_rating_soup() -> BeautifulSoup:
```

File: pipeline/extract/steam/conftest.py [hunk 14]
```diff
@@ -5,43 +5,51 @@
...
-def steam_no_rating_soup():
+def steam_no_rating_soup() -> BeautifulSoup:
+    '''Returns an example of html for no reviews.'''
     return BeautifulSoup(""" <div class="review_ctn"> 
                         "There are no reviews for this product"
                          </div>""", features='html.parser')
 
 
 @pytest.fixture
 def steam_rating_soup_positive_only():
```
Comment (annalisev) [inline pipeline/extract/steam/conftest.py:25]:
```suggestion
def steam_rating_soup_positive_only() -> BeautifulSoup:
```

File: pipeline/extract/steam/conftest.py [hunk 15]
```diff
@@ -5,43 +5,51 @@
...
+def steam_rating_soup_positive_only() -> BeautifulSoup:
+    '''Returns an example of html for the number of reviews,
+    when there are only positive reviews.'''
     return BeautifulSoup(""" <div class="review_ctn"> 
                          <label for="review_type_positive">Positive&nbsp;<span class="user_reviews_count">(105)</span></label><br> 
                          </div>""", features='html.parser')
 
 
 @pytest.fixture
 def steam_rating_soup_negative_only():
```
Comment (annalisev) [inline pipeline/extract/steam/conftest.py:34]:
```suggestion
def steam_rating_soup_negative_only() -> BeautifulSoup:
```

File: pipeline/extract/steam/conftest.py [hunk 16]
```diff
@@ -5,43 +5,51 @@
...
+def steam_rating_soup_negative_only() -> BeautifulSoup:
+    '''Returns an example of html for the number of reviews,
+    when there are only negative reviews.'''
     return BeautifulSoup(""" <div class="review_ctn"> 
                          <label for="review_type_negative">Negative&nbsp;<span class="user_reviews_count">(103)</span></label>
                          </div>""", features='html.parser')
 
 
 @pytest.fixture
 def steam_tags_good():
```
Comment (annalisev) [inline pipeline/extract/steam/conftest.py:43]:
```suggestion
def steam_tags_good() -> BeautifulSoup:
```

File: pipeline/extract/steam/conftest.py [hunk 17]
```diff
@@ -5,43 +5,51 @@
...
-def steam_tags_good():
+def steam_tags_good() -> BeautifulSoup:
+    '''Returns an example of html for game tags.'''
     return BeautifulSoup(""" <a href="https://store.steampowered.com/tags/en/Horror/?snr=1_5_9__409" class="app_tag" style="display: none;"> Horror	</a>
                          <a href="https://store.steampowered.com/tags/en/Atmospheric/?snr=1_5_9__409" class="app_tag" style="display: none;">Atmospheric</a>""",
                          features='html.parser')
 
 
 @pytest.fixture
 def steam_tags_none():
```
Comment (annalisev) [inline pipeline/extract/steam/conftest.py:51]:
```suggestion
def steam_tags_none() -> BeautifulSoup:
```

File: pipeline/extract/steam/conftest.py [hunk 18]
```diff
@@ -50,6 +58,7 @@ def steam_tags_none():
 
 @pytest.fixture
 def steam_developer_good():
```
Comment (annalisev) [inline pipeline/extract/steam/conftest.py:60]:
```suggestion
def steam_developer_good() -> BeautifulSoup:
```

File: pipeline/extract/steam/conftest.py [hunk 19]
```diff
@@ -60,6 +69,8 @@ def steam_developer_good():
 
 @pytest.fixture
 def steam_developer_none():
```
Comment (annalisev) [inline pipeline/extract/steam/conftest.py:71]:
```suggestion
def steam_developer_none() -> BeautifulSoup:
```

File: pipeline/extract/steam/conftest.py [hunk 20]
```diff
@@ -90,6 +102,7 @@ def steam_publisher_good():
 
 @pytest.fixture
 def search_result():
```
Comment (annalisev) [inline pipeline/extract/steam/conftest.py:104]:
```suggestion
def search_result() -> BeautifulSoup:
```

**Comment by annalisev** [comment]
I have added typehints, let me know if I should change anything else!

File: pipeline/extract/epic/conftest.py [hunk 21]
```diff
@@ -1,8 +1,11 @@
+'''A conftest file that defines functions used in test_epic.py.'''
+
 import pytest
 
 
 @pytest.fixture
-def epic_game_post_response():
+def epic_game_post_response() -> Dict:
```
Comment (annalisev) [inline pipeline/extract/epic/conftest.py:7]:
```suggestion
def epic_game_post_response() -> dict:
```

File: pipeline/extract/epic/conftest.py [hunk 22]
```diff
@@ -63,7 +66,8 @@ def epic_game_post_response():
 
 
 @pytest.fixture
-def epic_game_data_game_obj():
+def epic_game_data_game_obj() -> Dict:
```
Comment (annalisev) [inline pipeline/extract/epic/conftest.py:69]:
```suggestion
def epic_game_data_game_obj() -> dict:
```

File: pipeline/extract/epic/conftest.py [hunk 23]
```diff
@@ -115,7 +119,8 @@ def epic_game_data_game_obj():
 
 
 @pytest.fixture
-def epic_empty_post_request():
+def epic_empty_post_request() -> Dict:
```
Comment (annalisev) [inline pipeline/extract/epic/conftest.py:122]:
```suggestion
def epic_empty_post_request() -> dict:
```

File: pipeline/extract/steam/conftest.py [hunk 24]
```diff
@@ -59,15 +68,18 @@ def steam_developer_good():
 
 
 @pytest.fixture
-def steam_developer_none():
+def steam_developer_none() -> BeautifulSoup:
+    '''Returns an example of html for the developer,
```
Comment (hhm970) [inline pipeline/extract/steam/conftest.py:72]: Excellent detail in docstrings

**Review by hhm970 — APPROVED** [review]
I see nothing wrong with the docstrings and type hints you've put in; well-documented code is good code!

**Pull request merged**